func.go

```diff
@@ -133,6 +139,23 @@ func RegisterFunc(fptr interface{}, cfn uintptr) {
 				} else {
 					stack++
 				}
+			case reflect.Struct:
+				if runtime.GOOS != "darwin" || (runtime.GOARCH != "amd64" && runtime.GOARCH != "arm64") {
+					panic("purego: struct arguments are only supported on darwin amd64 & arm64")
+				}
+				addInt := func(u uintptr) {
+					ints++
+				}
+				addFloat := func(u uintptr) {
+					floats++
+				}
+				addStack := func(u uintptr) {
+					stack++
+				}
+				if arg.Size() == 0 {
+					continue
+				}
+				_ = addStruct(reflect.New(arg).Elem(), &ints, &floats, &stack, addInt, addFloat, addStack, nil)
```

Comment:
What about returning the counts of ints, floats and stacks instead of passing closures?

Reply:
Closures are needed for actually placing values into the registers down below. This is because addInt may place it on the stack if all the registers are full. It may be possible to return the ints, floats, stack instead of using pointers but is that as clear?

Reply:
We could but would we return just the new ints, floats, stack variables or keep them pointers in the struct?

Reply:
I think we would keep them pointers in the struct.

Reply:
Do we gain anything then by putting that state together?

Reply:
Readability would be improved (at least to me). What do you think?

Reply:
Isn't this `argumentState` as this is used not only for structs but other argument types?

Reply:
Well it's only passed around for structs right?

Reply:
My intention is replacing

```go
		var ints int
		var floats int
		var stack int

// ...

				addInt := func(u uintptr) {
					ints++
				}
				addFloat := func(u uintptr) {
					floats++
				}
				addStack := func(u uintptr) {
					stack++
				}
```

with

```go
type argumentState struct {
    ints int
    floats int
    stack int
}

func (a *argumentState) addInt(u uintptr) {
    a.ints++
}

func (a *argumentState) addFloat(u uintptr) {
    a.floats++
}

func (a *argumentState) addStack(u uintptr) {
    a.stack++
}

// We might be able to add the 'save' behavior as a method.

// ...

var state argumentState
```

and passing a pointer to `state` instead of 6 arguments. Would this make sense?

Reply:
On second thought, I found that addInt/addFloat/addStack's implementation differ, so if we would do, we would have to use an interface like this?

```go
type argumentState interface {
     addInt(u uintptr)
     addFloat(u uintptr)
     addStack(u uintptr)
}
```

As this seems a little big refactoring, it's OK not to do this in this PR. Let's leave the current arguments as they are.